docs/unity/part-2a-rust.md

```diff
@@ -0,0 +1,316 @@
+
+---
+
+### Publishing a Module to SpacetimeDB
+
+Now that we've written the code for our server module and reached a clean checkpoint, we need to publish it to SpacetimeDB. This will create the database and call the init reducer. In your terminal or command window, run the following commands.
+
+```bash
+cd server
+spacetime publish -c unity-tutorial
+```
+
+If you get any errors from this command, double check that you correctly entered everything into `lib.rs`. You can also look at the [Client Troubleshooting](/docs/unity/part-3.md#Troubleshooting) section.
+
+### Finally, Add Chat Support
+
+The client project has a chat window, but so far, all it's used for is the message of the day. We are going to add the ability for players to send chat messages to each other.
+
+First lets add a new `ChatMessage` table to the SpacetimeDB module. Add the following code to ``lib.rs``.
+
+**Append to the bottom of server/src/lib.rs:**
+
+```rust
+#[spacetimedb(table)]
+pub struct ChatMessage {
+    // The primary key for this table will be auto-incremented
+    #[primarykey]
+    #[autoinc]
+    pub message_id: u64,
+
+    // The entity id of the player that sent the message
+    pub sender_id: u64,
+    // Message contents
+    pub text: String,
+}
+```
+
+Now we need to add a reducer to handle inserting new chat messages.
+
+**Append to the bottom of server/src/lib.rs:**
+
+```rust
+// Adds a chat entry to the ChatMessage table
+#[spacetimedb(reducer)]
+pub fn send_chat_message(ctx: ReducerContext, text: String) -> Result<(), String> {
+    if let Some(player) = PlayerComponent::filter_by_owner_id(&ctx.sender) {
+        // Now that we have the player we can insert the chat message using the player entity id.
+        ChatMessage::insert(ChatMessage {
+            // this column auto-increments so we can set it to 0
+            message_id: 0,
+            sender_id: player.entity_id,
+            text,
+        })
+        .unwrap();
+
+        return Ok(());
+    }
+
+    Err("Player not found".into())
+}
+```
+
+## Wrapping Up
+
+Now that we added chat support, let's publish the latest module version to SpacetimeDB, assuming we're still in the `server` dir:
+
+```bash
+spacetime publish -c unity-tutorial
+```
+
+If you get any errors from this command, double check that you correctly entered everything into `lib.rs`. You can also look at the [Client Troubleshooting](/docs/unity/part-3.md#Troubleshooting) section.
+
+From here, the tutorial continues with more-advanced topics. The [next tutorial](/docs/unity/part-4.md) introduces Resources & Scheduling.
```

Comment:
I feel like this suggests that users should skip to part 4 instead unless they're having issues with the client, is that right? But pt 3 is still the next suggested step either way, no?

Maybe we could clarify that pt 4 continues with more advanced server topics, but pt 3 goes into integrating these into the client? Is that right?

Reply:
Got it: See the last 2 commits. The original doc bounced around, but the new one split client:server sections -- just legacy that didn't move.

The troubleshooting errs shouldn't actually show since it links to *client* troubleshooting (no answers there).

Fixed in the last 2 commits